src/permutation.rs

```diff
@@ -10,7 +10,7 @@ use poseidon252::sponge;
 
 /// Hashes a JubJub's ExtendedPoint into a JubJub's Scalar
 pub fn hash(p: &JubJubExtended) -> JubJubScalar {
-    let h = sponge::hash(&p.to_hash_inputs());
+    let h = sponge::hash(&p.to_hash_inputs()[..]);
```

Comment:
Why is this change necessary?

Reply:
As mentioned in our chat my compiler wasn't updated to the version where this is no longer necessary. I rollback the change and also updated where it was used elsewhere.